packages/frontend/src/features/catalog/components/CatalogPanel.tsx

```diff
@@ -123,18 +124,18 @@ export const CatalogPanel: FC = () => {
 
     const { mutate: getMetadata } = useCatalogMetadata(projectUuid, (data) => {
         if (data) {
-            setMetadata(data);
-        }
-
-        if (!isSidebarOpen && data) {
-            setSidebarOpen(true);
+            startTransition(() => {
+                setMetadata(data);
+            });
         }
     });
     const { mutate: getAnalytics } = useCatalogAnalytics(
         projectUuid,
         (data) => {
             if (data) {
-                setAnalyticsResults(data);
+                startTransition(() => {
```

Comment:
I'm not exactly sure what's happening here, but my guess is that this analytics can take longer than `metadata` request . And we should not block the metadata panel while this analytic results is being requested, just so you know. We could add a `delay` on the backend to simulate what might happen.